packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -288,7 +293,36 @@ export class StripeWebhookHandler extends StripeHandler {
     if (customer.deleted) {
       return;
     }
-    return this.paypalHelper.processInvoice({ customer, invoice });
+    try {
+      await this.paypalHelper.processInvoice({
+        customer,
+        invoice,
+        batchProcessing: true,
+      });
+      return true;
+    } catch (err) {
+      if (err instanceof PayPalClientError) {
+        if (err.errorCode === PAYPAL_BILLING_AGREEMENT_INVALID) {
+          const uid = customer.metadata.userid;
+          const billingAgreementId = this.stripeHelper.getCustomerPaypalAgreement(
+            customer
+          ) as string;
+          await this.stripeHelper.removeCustomerPaypalAgreement(
+            uid,
+            customer.id,
+            billingAgreementId
+          );
+          await this.sendSubscriptionPaymentFailedEmail(invoice);
```

Comment:
If we remove the PayPal BA and keep the subscription, what does the customer see in the subscription management page?  Maybe @biancadanforth or @meandavejustice know.

Reply:
They should see a prompt to put a new PayPal agreement on file. There's a specific response to the `GET .../customer` I added that the front-end will use to determine whether to show it.